src/api.js

```diff
@@ -0,0 +1,48 @@
+
+export async function apiGet(url) {
+    try {
+        const response = await fetch(url);
+        if (!response.ok) {
+            throw new Error(`Erro ao buscar dados de ${url}`);
+        }
+        return await response.json();
+    } catch (error) {
+        throw new Error(error.message);
+    }
+}
+
+export async function apiPost(url, body) {
+    try {
+        const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(body)
+        });
+        if (!response.ok) {
+            throw new Error(`Erro ao postar dados em ${url}`);
+        }
```

Comment:
Cara, você vê necessidade nisso aqui?
Parece ambíguo.  Se não for OK ele já vai cair no try catch.


```
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(body)
        });
   return await response.json();
    } catch (error) {
        throw new Error(error.message);
    }
```